### ISSUE TITLE
Downlod to Tolino vsion 4 HD not successful!

### ISSUE THREAD

**Issue by tolino4hduser** [rvolz/BicBucStriim#239]
Problem: 
As written in the headline the download of ebooks (in epub format) to my tolino vision 4 HD (SW V1.9.2) is not successful/possible. 

Error message: 

> Beim Laden ihrer Datei ist ein Fehler aufgetreten. Die Datei ist vermutlich beschädigt und kann nicht geöffnet werden.

Add Info: 
Installation of bbs  V1.3.6 on a Synology NAS with the SW DSM 6.xx successful completed, incl. check of a valid Calibre library and ebooks in epub-format. Everything looks fine so far. 
Login protection disabled / switch of the Authentication at Download, because of the note: "_Some ebook readers, like the Tolino, can’t download ebooks while the login protection is in effect_."
Going with the browser of my tolino to https://<nas>/bbs works fine and I can see  the books, but after clicking Download and selecting the epub format, I get the above mentioned error message. Copying the same ebook via USB cable on the Tolino is successful and I can read it. 
Note: This ebook is NOT protected by a DRM.
On my old Kindle Touch the download of this ebook is working and I can read it, of cause using the mobi ebook format. 

Do you have any ideas, what to check to allocate the root cause of the problem?
Or even better do you have a solution for this problem? 

In advance thanks for your support!

**Comment by rvolz** [comment]
When I last checked (see #114) Tolinos were able to download when the login protection was disabled, but that was a year ago. I'll try to get a Tolino and see what is going on. 

In the meantime you could turn on debug mode and check if there are any errors on the server side, that could cause this behavior.  https://github.com/rvolz/BicBucStriim/wiki/Debug-mode

**Comment by tolino4hduser** [comment]
Hallo Herr Volz,

vielen Dank für die prompte Reaktion. Super!
Ihr Programm ist wirklich genial.
Und sorry, daß ich sie mit meiner Tolino Fehlermeldung nerve,
aber vielleicht gibt es ja doch noch eine Chance das Problem zu lösen. :-)

Habe den Debug mode angeschaltet und mal zwei nicht erfolgreiche Download-Versuche (eimal als epub und einmal als pdf) gestartet.
Anbei der Inhalt der Log-Datei „2017-01-31.log“ dazu:

INFO - 2017-01-31T21:56:00+01:00 - BicBucStriim 1.3.6: Running in debug mode.
DEBUG - 2017-01-31T21:56:00+01:00 - easy mode: login not required
DEBUG - 2017-01-31T21:56:00+01:00 - own_config_middleware: config loaded
DEBUG - 2017-01-31T21:56:00+01:00 - login resource: /
INFO - 2017-01-31T21:56:02+01:00 - BicBucStriim 1.3.6: Running in debug mode.
INFO - 2017-01-31T21:56:02+01:00 - BicBucStriim 1.3.6: Running in debug mode.
DEBUG - 2017-01-31T21:56:02+01:00 - easy mode: login not required
DEBUG - 2017-01-31T21:56:02+01:00 - own_config_middleware: config loaded
DEBUG - 2017-01-31T21:56:02+01:00 - login resource: /titles/6/thumbnail
DEBUG - 2017-01-31T21:56:02+01:00 - thumbnail: 6
DEBUG - 2017-01-31T21:56:02+01:00 - easy mode: login not required
DEBUG - 2017-01-31T21:56:02+01:00 - own_config_middleware: config loaded
DEBUG - 2017-01-31T21:56:02+01:00 - login resource: /titles/7/thumbnail
DEBUG - 2017-01-31T21:56:02+01:00 - thumbnail: 7
INFO - 2017-01-31T21:56:09+01:00 - BicBucStriim 1.3.6: Running in debug mode.
DEBUG - 2017-01-31T21:56:09+01:00 - easy mode: login not required
DEBUG - 2017-01-31T21:56:09+01:00 - own_config_middleware: config loaded
DEBUG - 2017-01-31T21:56:09+01:00 - login resource: /tagslist/0/
INFO - 2017-01-31T21:56:15+01:00 - BicBucStriim 1.3.6: Running in debug mode.
DEBUG - 2017-01-31T21:56:15+01:00 - easy mode: login not required
DEBUG - 2017-01-31T21:56:15+01:00 - own_config_middleware: config loaded
DEBUG - 2017-01-31T21:56:15+01:00 - login resource: /tags/3/0/
INFO - 2017-01-31T21:56:21+01:00 - BicBucStriim 1.3.6: Running in debug mode.
DEBUG - 2017-01-31T21:56:21+01:00 - easy mode: login not required
DEBUG - 2017-01-31T21:56:21+01:00 - own_config_middleware: config loaded
DEBUG - 2017-01-31T21:56:21+01:00 - login resource: /titles/2
DEBUG - 2017-01-31T21:56:22+01:00 - titleDetails custom columns: 0
INFO - 2017-01-31T21:56:24+01:00 - BicBucStriim 1.3.6: Running in debug mode.
DEBUG - 2017-01-31T21:56:24+01:00 - easy mode: login not required
DEBUG - 2017-01-31T21:56:24+01:00 - own_config_middleware: config loaded
DEBUG - 2017-01-31T21:56:24+01:00 - login resource: /titles/2/cover/
INFO - 2017-01-31T21:56:29+01:00 - BicBucStriim 1.3.6: Running in debug mode.
DEBUG - 2017-01-31T21:56:29+01:00 - easy mode: login not required
DEBUG - 2017-01-31T21:56:29+01:00 - own_config_middleware: config loaded
DEBUG - 2017-01-31T21:56:29+01:00 - login resource: /tags/3/0/
INFO - 2017-01-31T21:58:00+01:00 - BicBucStriim 1.3.6: Running in debug mode.
DEBUG - 2017-01-31T21:58:00+01:00 - easy mode: login not required
DEBUG - 2017-01-31T21:58:00+01:00 - own_config_middleware: config loaded
DEBUG - 2017-01-31T21:58:00+01:00 - login resource: /tagslist/0/
INFO - 2017-01-31T21:58:09+01:00 - BicBucStriim 1.3.6: Running in debug mode.
DEBUG - 2017-01-31T21:58:09+01:00 - easy mode: login not required
DEBUG - 2017-01-31T21:58:09+01:00 - own_config_middleware: config loaded
DEBUG - 2017-01-31T21:58:09+01:00 - login resource: /tags/3/0/

Ist es das, was sie meinten?


Beste Grüße

Dirk Brucks

#################################

Am 31.01.2017 um 21:27 schrieb Rainer Volz <notifications@github.com>:

When I last checked (see #114) Tolinos were able to download when the login protection was disabled, but that was a year ago. I'll try to get a Tolino and see what is going on.

In the meantime you could turn on debug mode and check if there are any errors on the server side, that could cause this behavior. https://github.com/rvolz/BicBucStriim/wiki/Debug-mode

**Comment by tolino4hduser** [comment]
Meanwhile I have updated my tolino firmware to the latest software 1.9.4, but the problem still exists.
Any news? Did you get a tolino to test so far?
Br,
tolino4hduser

**Comment by rvolz** [comment]
Finally, I was able to test a Tolino 4 HD with Firmware 1.9.4, and to my surprise now everything worked fine. I used an installation with login, and I got a warning about an unrecognizable format when downloading -- maybe it doesn't like the EPUB mime type we send? -- but the downloaded file was ok and usable.

Did you try to switch the *metadata update* feature on and off? A while ago there was an issue with bad metadata that corrupted the download.

**Comment by rvolz** [comment]
... And the warning was due to problem in the book, not all of its files were well-formed. Other titles downloaded without the warning. 

So you might want to check problematic books with [Sigil](https://sigil-ebook.com/). It tells you if there are problems inside, that Calibre can deal with, but the Tolino cannot.

**Comment by tolino4hduser** [comment]
Today I received an other Tolino, this time a vision 3 HD (firmware 1.9.4) and it has the same bad behavior as described above and ended up with a white page and no epub file on my tolino. :-(

I tried to switch the _metadata update_ feature on and off, but no improvement.

Copying the "original" epub ebook(s) (with calibre converted) direkt from Mac via USB cable to the tolino works fine. Downloading the ebooks in safari browser via bbs on my mac and copying this downloaded ebook from Mac to the tolino via USB cable works fine as well.  
So in principle the downloaded ebooks via bbs are readable with the tolino, but using the tolino browser it won't work with my tolinos.

How to check the meta data or even headers of an epub ebook with Sigil?
What to look at? How should it be normal?
Sorry to ask you all this questions, but I load a ebook with Sigil editor, but I don't know what to look at.
Thank you so far.
tolinoxhduser

**Comment by tolino4hduser** [comment]
I made some more tests and I found a (the) reason, why downloading was not successful.
For the connection to my synology NAS I used "https", when the download of the ebooks was not successful. When using "http" only, download succeed!
At least "http" only has to be used downloading with the browser of a tolino.
On my Mac "https" in the Safari browser works as well.

May be this information helps others too.
